src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -90,17 +90,16 @@ static bool KeyExists(FabricIndex fabricIndex, GroupId groupId)
 
     auto it    = provider->IterateGroupKeys(fabricIndex);
     bool found = false;
-    while (it->Next(entry) && !found)
+    while (!found && it->Next(entry))
```

Comment:
Why did no CI test catch these? Are we unable to run tests for groups?

It would be nice if this PR would be accompanied by a test that makes sure we never fail again.